backend/utils/volunteer-sheet-parser.js

```diff
@@ -0,0 +1,84 @@
+// run `node utils/volunteer-sheet-parser.js <insert_spreadsheet_path>` from the `/backend` folder
+
+const XLSX = require("xlsx");
+const extractor = require("keyword-extractor");
+const resourceUtils = require("./resource-utils");
+const Resource = require("../models/Resource");
+const mongoose = require("mongoose");
+
+mongoose.connect(process.env.DB_URI, {
+  useUnifiedTopology: true,
+  useNewUrlParser: true
+});
+
+const formatNotes = ({
+  "Form Received": formReceived,
+  Availability: availability,
+  "How did you hear about LAH?": refer,
+  "Volunteer Roles": roles
+}) =>
+  `Form Received: ${formReceived}\n\n` +
+  `Availability: ${availability}\n\n` +
+  `How they heard about LAH: ${refer}\n\n` +
+  `Volunteer Roles: ${roles}\n\n`;
+
+const createTags = ({
+  "18 or Older?": eighteen,
+  "Skills, Qualifications, Current Occupation": skills,
+  "Do you have a degree in the mental health field?": mentalHealth,
+  "Volunteer Roles": roles,
+  "Willing to travel?": travel
+}) => [
+  eighteen === "Yes" ? "18+" : null,
+  mentalHealth === "Yes" ? "Mental Health" : null,
+  travel === "Yes" ? "Able to travel" : null,
+  ...extractor.extract(`${skills} ${roles}`, {
+    language: "english",
+    remove_digits: true,
+    return_changed_case: true,
+    remove_duplicates: true
+  })
+];
+
+const getLocation = async mailingAddress => {
+  try {
+    const { lat, lng, region, ...address } = await resourceUtils.geocodeAddress(mailingAddress);
+    return {
+      location: {
+        coordinates: [lng, lat]
+      },
+      federalRegion: region,
+      address
+    };
+  } catch (err) {
+    return {
+      location: {
+        coordinates: [null, null]
+      },
+      federalRegion: -1
+    };

```

Comment:
should i put a fake address here so it doesn't fail and then you can comb through and make sure the fake address isn't there? or should I leave this as is?

Reply:
I would be inclined to have the system stay fail-fast—what do you think though?

Reply:
I'd probably agree. I'll just have it log here, so that it's easy to find the source of the issue.

Reply:
updated